src/services/share-buttons.service.ts

```diff
@@ -0,0 +1,171 @@
+                return this.pinCount(url, count$);
+            default:
+                return;
+        }
+    }
+
+    private fbCount(url: string, count$: EventEmitter<number>) {
+        this.fetch(`https://graph.facebook.com/?id=${url}`)
+            .subscribe((data: any) => {
+                data = data.json();
+                if (data.hasOwnProperty('share') && data.share.hasOwnProperty('share_count')) {
+                    count$.emit(data.share.share_count);
+                    return;
+                }
+                count$.emit(0);
+            });
+    }
+
+    private linkedInCount(url: string, count$: EventEmitter<number>) {
+        this.fetchJsonp(`https://www.linkedin.com/countserv/count/share?url=${url}`)
+            .subscribe((data: any) => {
+                data = data.json();
+                count$.emit(data.count || 0);
+            });
+    }
+
+    private redditCount(url: string, count$: EventEmitter<number>) {
+        this.fetch(`https://buttons.reddit.com/button_info.json?url=${url}`)
+            .subscribe((data: any) => {
+                data = data.json();
+                if (data.hasOwnProperty('data') && data.data.hasOwnProperty('children')) {
+                    if (data.data.children.length) {
+                        count$.emit(data.data.children[0].data.score);
+                        return;
+                    }
+                }
+                count$.emit(0);
+            });
+    }
+
+    private gPlusCount(url: string, count$: EventEmitter<number>) {
+        let body = Helper.gplusCountBody(url);
+        this.post('https://clients6.google.com/rpc?key=AIzaSyCKSbrvQasunBoV16zDH9R33D88CeLr9gQ', body)
+            .subscribe((data: any) => {
+                data = data.json();
+                if (data[0] && data[0].hasOwnProperty('result')) {
+                    count$.emit(data[0].result.metadata.globalCounts.count);
+                    return;
+                }
+                count$.emit(0);
+            });
+    }
+
+    private pinCount(url: string, count$: EventEmitter<number>) {
+        this.fetch(`https://api.pinterest.com/v1/urls/count.json?callback=receiveCount&url=${url}`)
+            .subscribe((data: any) => {
+                data = data.text();
+                let result = JSON.parse(data.replace(/^receiveCount\((.*)\)/, '$1'));
+                count$.emit(result.count || 0);
+            });
+    }
+
+    private tumblrCount(url: string, count$: EventEmitter<number>) {
+        this.fetchJsonp(`https://api.tumblr.com/v2/share/stats?url=${url}`)
+            .subscribe((data: any) => {
+                data = data.json();
+                if (data.hasOwnProperty('response') && data.response.hasOwnProperty('note_count')) {
+                    count$.emit(data.response.note_count);
+                    return;
+                }
+                count$.emit(0);
+            });
+    }
+
+    private post(url: string, body: any) {
```

Comment:
Parameterize the `post`, `fetch`and `fetchJsonp` methods to improve intellisense:
```
 private post<T>(url: string, body: any): Observable<T> {
         let headers = new Headers({ 'Content-Type': 'application/json' });
         let options = new RequestOptions({ headers: headers });
         return this.http.post(url, body, options)
             .catch((err) => Observable.empty<T>());
     }
```